File: src/archive/CSESJava/src/main/java/org/allan7yin/intro/IncreasingArray.java
package org.allan7yin.intro;

import java.util.ArrayList;
import java.util.Scanner;

public class IncreasingArray {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();

        ArrayList<Long> nums = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            nums.add(scanner.nextLong());
        }

        long cost = 0;
        long prev = -1;

        for (int i = 0; i < n; i++) {
            long current = nums.get(i);
            if (current < prev) {
                cost += (prev - current);
                nums.set(i, prev);
            }
            prev = nums.get(i);
        }

        System.out.println(cost);

        scanner.close();
    }
}

File: src/archive/CSES/Intro/increasing_array.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
    int n;
    cin >> n;
    vector<long long> nums;
    for (int i = 0; i < n; i++) {
        long long num;
        cin >> num;
        nums.push_back(num);
    }

    long long cost = 0;
    int prev = -1;
    for (int i = 0; i < n; i++) {
        if (nums[i] < prev) {
            cost += (prev - nums[i]);
            nums[i] = prev;
        }
        prev = nums[i];
    }
    cout << cost << endl;
}